src/main/java/com/solvd/RentalCompany/services/ReservationService.java

```
@@ -0,0 +1,59 @@
+package com.solvd.RentalCompany.services;
+
+import com.solvd.RentalCompany.core.ReservationCore;
+import com.solvd.RentalCompany.dao.IReservationDAO;
+import com.solvd.RentalCompany.dao.ReservationDAO;
+import com.solvd.RentalCompany.exceptionHandlers.ObjectNotFoundException;
+import com.solvd.RentalCompany.mappers.IReservationMapper;
+import com.solvd.RentalCompany.mappers.ReservationMapper;
+import com.solvd.RentalCompany.models.Reservation;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import java.util.Optional;
+
+public class ReservationService implements IReservationService{
+    private final static Logger logger = LogManager.getLogger(ReservationService.class);
+    private final IReservationDAO reservationDAO = new ReservationDAO();
+    private final IReservationMapper reservationMapper = new ReservationMapper();
+    @Override
+    public ReservationCore createEntity(ReservationCore reservationCore) {
```

lets overload this with
```
public ReservationCore createEntity(Client client, Car car, Employee employee, Insurance insurance, Date dateFrom, Date dateTo);
public ReservationCore createEntity(Client client, Car car, Employee employee, Insurance insurance, Date dateTo) // suggesting dateFrom is right now
public ReservationCore createEntity(Client client, Car car, Employee employee, Insurance insurance) // suggesting dateFrom is right now and dateTo is unknown and will be set once client returnes the car
```

---

I overloaded this method, everything works, but it turned out quite hardcoded, I think. Surely you know a better way to do this.

---

`Car, Employee and InsuranceType should only be set from the existing one` if you use REFERENCES keqyword while creating database, you will get an error while insert, and then just catch it and work it out. 
 `the Client can be new or already existing in the database` you should decide what do you mean by `already exist` and once again you can throw an error if user not exists. Because creating a user while creating a reservation feels strange to me.